Insert fails when entering rebalance kick loop

In cfiter.go, the Insert method fails during rebalance with an index calculated to be greater than the the size of the filter. The index should be equal to or less than the max size of the filter.

Script to reproduce the error condition.

```
func main() {
	cf := cfilter.New(cfilter.FingerprintSize(3), cfilter.Size(10))
	for i := 0; i < 29; i++ { // 28 will work 29 ... 40 will fail

		cf.Insert([]byte("hello, fir" + strconv.Itoa(i+3) + "+st timer" + strconv.Itoa(i) + "aa"))
	}
	cf.Dump() // visualize filter
	fmt.Println(cf.Count())
}

// visualize contents
func (cf *CFilter) Dump() {
	for a, b := range cf.buckets {
		fmt.Print(a)
		for _, c := range b {
			fmt.Print("[", hex.EncodeToString(c), "]")
		}
		fmt.Println()
	}
}

```

When all the buckets are full, execution will drop down to the rebalance section and do at max 500 kicks, however it crashes early with the following error: 

panic: runtime error: index out of range
.../cfilter.go:64 +0x786

Introducing the following commented code shows i (before and after the % mod). Notice that i is calculated to 13 however the filter size is 10, so it will panic with an index out of range error and will not do the kick and rebalance.

```
// Insert adds an element (in byte-array form) to the Cuckoo filter,
// returns true if successful and false otherwise.
func (cf *CFilter) Insert(item []byte) bool {
	f := fprint(item, cf.fpSize, cf.hashfn)
	j := hashfp(item) % cf.size
	k := (j ^ hashfp(f)) % cf.size

	if cf.buckets[j].insert(f) || cf.buckets[k].insert(f) {
		cf.count++
		return true
	}

	i := [2]uint{j, k}[rand.Intn(2)]
	for n := uint(0); n < cf.kicks; n++ {
		f = cf.buckets[i].swap(f)
		fmt.Println(hex.EncodeToString(f), j, k, i) // show the values
		i ^= hashfp(f) % cf.size
		fmt.Println(hex.EncodeToString(f), j, k, i) // show the values

		if cf.buckets[i].insert(f) {
			cf.count++
			return true
		}
	}

	return false
```

Sample output.
Running with 28 inserts works because no kicks happen.

> 0[fc84f2][48f321][36bd37][]
> 1[09f92d][65de72][87fa10][32ed51]
> 2[069309][b8a269][69c550][e4ec1f]
> 3[0a2d31][83995a][][]
> 4[9cd789][be9358][eb6ed0][]
> 5[c955b3][][][]
> 6[77a10b][6bf9c9][371ed3][]
> 7[99cb92][5784ec][][]
> 8[**92c257**][54d3ef][**edcb06**][beb287]
> 9[6cafbb][8aa567][][]
> 28

Running with 29 inserts (falls into kick rebalance section) and crashes because 13 is calculated index, but 10 is filter size:

> **92c257** 1 8 8
> 92c257 1 8 8
> **edcb06** 1 8 8
> edcb06 1 8 **13**
> panic: runtime error: index out of range
> ...cfilter.go:66 +0x786

Setting fingerprints to 8 will the cause same failure to occur however it happens at insert number 34, but this time the buckets j,k were equal.

```
0[fc84f2e7ca2d5e96][48f321721bda24b2][36bd3783fab8ee80][8b4daa6ff23af8ee]
1[09f92dfded564d0b][65de720dfb99e4c9][87fa10567aab693f][1ba6b55473eb9b1f]
2[069309e6acb574ee][b8a269498459a140][69c55042abe22ab6][e4ec1f7924272d3c]
3[6cafbb55b62421f6][83995a56a985e5c7][][]
4[**9cd78960d30fbb5a**][be9358f5060bdf40][eb6ed07472612f90][6015587b8b8596fa]
5[0a2d31eb08846f48][32ed51cfd8d0f3ba][c955b301a3e8fe27][e584dc5296c215a7]
6[77a10b937c13ec62][6bf9c92e470384c6][371ed307bfc0bf10][8279d0bc9502ca72]
7[5784ec3b63a36bbf][][][]
8[92c25770e1271a9c][54d3ef79af523eae][edcb06ae5ff37f20][99cb926b6f4bce38]
9[8aa56766d6b5765f][beb287779de631b8][][]
33
```
With 34 inserts, failure occurs due to index calculation of 12.

> 9cd78960d30fbb5a 4 4 4
> 9cd78960d30fbb5a 4 4 **12**


The ultimate result of this error is that it when the kicks need to be invoked the program crashes thus preventing a graceful rebalance or on hitting 500 kicks, it should report that the filter is full. It is unclear to me how to rectify this even after reading the paper. The logic seems correct with Algorithm 1, so the issue appears to be related the hashfp() and % mod to get the filters bucket row number as this is calculating an index greater than cf.size.  This is a bug that breaks ability to maximize the load of the DB. I first noticed this with a POC with a cfilter sized to 1MM, and stepped it back to 10 to be able to visualize what what was happening.

--- a/cfilter.go
+++ b/cfilter.go
@@ -57,7 +57,7 @@ func (cf *CFilter) Insert(item []byte) bool {
 	i := [2]uint{j, k}[rand.Intn(2)]
 	for n := uint(0); n < cf.kicks; n++ {
 		f = cf.buckets[i].swap(f)
-		i ^= hashfp(f) % cf.size
+		i = (i ^ hashfp(f)) % cf.size
 
 		if cf.buckets[i].insert(f) {
 			cf.count++

